components/switch/switch-mixin.js

```diff
@@ -192,6 +195,7 @@ export const SwitchMixin = superclass => class extends FocusMixin(RtlMixin(Focus
 		const textPosition = (this.textPosition === 'start' || this.textPosition === 'hidden'
 			? this.textPosition : 'end');
 
+		// Note: we render the switchLabel in the case of textPosition === 'hidden' so that any slot handlers can pick up on content being passed in
```

Comment:
In this case, I'm specifically referring to the Visibility Switch's `_handleConditionsSlotChange`, which is necessary for the `_hasConditions` flag to be properly set.